### PR TITLE
fix(sdk): fix support for args when using IndexifyFunction and IndexifyRouter classes

### PR DESCRIPTION
**Pull request by seriousben**
## Context

We introduced a regression making this graph fail:

```py
        class MyObject(BaseModel):
            x: int

        class SimpleFunctionCtxCls(IndexifyFunction):
            name = "SimpleFunctionCtxCls"

            def __init__(self):
                super().__init__()

            def run(self, obj: MyObject) -> MyObject:
                return MyObject(x=obj.x + 1)

        graph = Graph(name="test_simple_function_cls", start_node=SimpleFunctionCtxCls)
        graph = remote_or_local_graph(graph, is_remote)
        invocation_id = graph.run(block_until_done=True, obj=MyObject(x=1))
        output = graph.output(invocation_id, "SimpleFunctionCtxCls")
        self.assertEqual(output, [MyObject(x=2)])
```

It failed because it called the run function with `dict(obj=MyObject(x=1))` instead of just `MyObject(x=1)`.

## What

Recent improvements to args and kwargs handling missed support for IndexifyFunction and IndexifyRouter classes.

This PR aligns the approach taken for function and classes.

## Testing

Add a new test case for IndexifyFunction and IndexifyRouter classes.

## Contribution Checklist

- [x] If the python-sdk was changed, please run `make fmt` in `python-sdk/`.
- [x] If the server was changed, please run `make fmt` in `server/`.
- [x] Make sure all PR Checks are passing.

### CODE REVIEW

**Comment by stangirala** [comment]
+1 (needs pr check fix)

File: python-sdk/indexify/function_executor/handlers/run_function/handler.py [hunk 1]
```diff
@@ -139,9 +139,10 @@ def _indexify_client(
 
 
 def _is_router(func_wrapper: IndexifyFunctionWrapper) -> bool:
-    return (
-        str(type(func_wrapper.indexify_function))
-        == "<class 'indexify.functions_sdk.indexify_functions.IndexifyRouter'>"
+    return str(
+        type(func_wrapper.indexify_function)
+    ) == "<class 'indexify.functions_sdk.indexify_functions.IndexifyRouter'>" or isinstance(
+        func_wrapper.indexify_function, IndexifyRouter
```
[the remark this answers is not in the document]
Comment (seriousben) [reply python-sdk/indexify/function_executor/handlers/run_function/handler.py:145]: From my testing, the way we build the IndexifyFunction using `type(...)` fails the is instance check.

Will do more testing.

[the remark this answers is not in the document]
Comment (eabatalov) [reply python-sdk/indexify/function_executor/handlers/run_function/handler.py:145]: Interesting. If isinstance() fails and we won't figure out why, then let's not add isinstance() here and add a comment that's it's not working. So others don't try to quickfix it with the isinstance() check.

[the remark this answers is not in the document]
Comment (seriousben) [reply python-sdk/indexify/function_executor/handlers/run_function/handler.py:145]: I would not be adding code for now reason :)

The isintance is needed when using the IndexifyRouter class. But it looks like when using the indexify_router function, it is not enough.

[the remark this answers is not in the document]
Comment (eabatalov) [reply python-sdk/indexify/function_executor/handlers/run_function/handler.py:145]: TIL :)

[the remark this answers is not in the document]
Comment (seriousben) [reply python-sdk/indexify/function_executor/handlers/run_function/handler.py:149]: Looks like it can either be an instance or a class.

**Pull request merged**